main/guides/integration/name-services.md

```diff
@@ -0,0 +1,106 @@
+# Name Services
+
+The Agoric chain has a few name services:
+ - `agoricNames` for well-known names controlled by governance
+ - `namesByAddress` - a namespace for each provisioned account
+ - `board` - to map objects to arbitrary names and back
+
+## agoricNames / agoricNamesAdmin - well known names
+
+To look up well-known objects, such as the BLD brand:
+
+```js
+const brandHub = E(agoricNames).lookup('brand');
+const bldBrand = await E(brandHub).lookup('BLD');
+```
+
+or, equivalently:
+
+```js
+const bldBrand = await E(agoricNames).lookup('brand', 'BLD');
+```
+
+`agoricNames` is a _NameHub_. In general, `E(hub).lookup(key1, key2, ...rest)` is equivalent
+to `E(E(hub).lookup(key1)).lookup(key2, ...rest)`. The NameHub interface also includes `Map`-like
+methods `has`, `entries`, `keys`, and `values`.
+
+The keys of `agoricNames` include:
+ - `brand` - IST, BLD, Invitation (Zoe invitation brand), timer (a TimerBrand, not an ERTP brand)
+ - `issuer` - IST, BLD, Invitation, ...
+ - `vbankAsset` - IST, BLD, USDC, ATOM, ... - including info about corresponding cosmos denom
+ - `oracleBrand` - USD, ATOM
+ - `installation` - psm, reserve, priceAuthority, ... (see [deployed contracts](../zoe/actual-contracts/))
+ - `instance` - reserve, one for each psm, one for each priceAuthority, ... (see [deployed contracts](../zoe/actual-contracts/))
+
+Adding or updating an entry requires access to the corresponding _NameAdmin_ facet: `agoricNamesAdmin` or
+one of its descendants - most likely as part of [permissioned contract deployment](../coreeval/). For example:
+
+```js
+const { instance } = await E(zoe).startInstance(myNewContract);
+const { brands: { Thing: thingBrand } } = await E(zoe).getTerms(instance);
+const brandAdmin = E(agoricNamesAdmin).lookupAdmin('brand');
+await E(brandAdmin).update('Thing', thingBrand);
```

Comment:
This is a _bad_ example. BLD staker governance should not approve meaningless names such as **Thing**.

Reply:
Also TODO: note the risk of registering under a name such as **BTC** a brand for an asset that I just made up and that the market will _not_ value at 1 BTC.